fe1-web/Model/Network/Method/Message/data/witness/WitnessMessage.ts

```diff
@@ -16,14 +16,17 @@ export class WitnessMessage implements MessageData {
         this.message_id = msg.message_id;
 
         if (!msg.signature) throw new ProtocolError('Undefined \'signature\' parameter encountered during \'WitnessMessage\'');
+        // FIXME verify signature without the public key available? 0.o + uncomment 3 tests in "FromJson" test suite
```

Comment:
Technically the public key should match the sender's, in the message containing the message data.
But this points to something we might want to fix in the protocol.